sstables/consumer.hh

```diff
@@ -114,8 +116,10 @@ private:
     } _read_int;
 
     // state for READING_BYTES prestate
+    size_t _read_bytes_len = 0;
     temporary_buffer<char> _read_bytes;
-    temporary_buffer<char>* _read_bytes_where; // which temporary_buffer to set, _key or _val?
+    std::vector<temporary_buffer<char>> _read_bytes_fragments;
+    std::variant<temporary_buffer<char>*, fragmented_temporary_buffer*> _read_bytes_where; // which buffer to set, _key, _val, _cell_path or _pk?
```

Comment:
Would it make sense to always use fragmented_temporary_buffer and don't have those special cases?

Reply:
There are still some cases where we want to parse bytes into a contiguous buffer(mostly keys), so instead of linearizing a fragmented_temporary_buffer in every such place, I've left optional methods for contiguous parsing.
At least, after the last changes, these don't look like 'special' cases, just like separate cases.